autoroutes.pyx

```diff
@@ -66,19 +119,27 @@ cdef class Edge:
         cdef:
             list parts
             str pattern, segment
+            unsigned int match_type
         if self.pattern_start != -1 and self.pattern_end != -1:
-            segment = self.pattern[self.pattern_start:self.pattern_end]
-            parts = segment.split(':')
-            if len(parts) == 2:
-                pattern = parts[1]
-            else:
-                pattern = DEFAULT_MATCH_TYPE
-            self.match_type = MATCH_TYPES.get(pattern, MATCH_REGEX)
+            pattern, self.match_type = Edge.extract_pattern(self.pattern[self.pattern_start:self.pattern_end])
         else:
             pattern = self.pattern
             self.match_type = 0  # Reset, in case of branching.
         return pattern
 
+    @staticmethod
```

Comment:
Why a `staticmethod` here? `pattern = self.extract_pattern()` which fill `self.math_type` looks fine to me.

Reply:
Good catch, it was a static method before I decided to move the code from Routes to Edge. Will fix.